src/dotnet-roslyn-tools/PRTagger/PRTagger.cs

```diff
@@ -0,0 +1,205 @@
+    }
+
+    private static bool TryGetProductInfo(string productName, [NotNullWhen(true)] out IProduct? productInfo)
+    {
+        if (productName.Equals(s_roslynInfo.Name, StringComparison.OrdinalIgnoreCase))
+        {
+            productInfo = s_roslynInfo;
+            return true;
+        }
+        else if (productName.Equals(s_razorInfo.Name, StringComparison.OrdinalIgnoreCase))
+        {
+            productInfo = s_razorInfo;
+            return true;
+        }
+
+        productInfo = null;
+        return false;
+    }
+
+    private static async Task<string?> TryGetBuildNumberForReleaseAsync(
+        IProduct productInfo,
+        string vsCommitSha,
+        AzDOConnection vsConnection)
+    {
+        var url = await VisualStudioRepository.GetUrlFromComponentJsonFileAsync(
+            vsCommitSha, GitVersionType.Commit, vsConnection, productInfo.ComponentJsonFileName, productInfo.ComponentName);
+        if (url is null)
+        {
+            return null;
+        }
+
+        try
+        {
+            var buildNumber = VisualStudioRepository.GetBuildNumberFromUrl(url);
+            return buildNumber;
+        }
+        catch
+        {
+            return null;
+        }
+    }
+
+    private static async Task<string?> TryGetProductCommitShaFromBuildAsync(
+        IProduct productInfo,
+        AzDOConnection buildConnection,
+        string buildNumber)
+    {
+        var build = (await buildConnection.TryGetBuildsAsync(
+            productInfo.GetBuildPipelineName(buildConnection.BuildProjectName)!, buildNumber))?.SingleOrDefault();
+
+        return build?.SourceVersion;
+    }
+
+    private async static Task<bool> TryCreateIssue(
+        IProduct product,
+        string vsBuildNumber,
+        string issueBody,
+        string gitHubToken,
+        ILogger logger)
+    {
+        var client = new HttpClient
+        {
+            BaseAddress = new("https://api.github.com/")
+        };
+
+        var authArray = Encoding.ASCII.GetBytes($"{gitHubToken}");
+        client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(
+            "Basic",
+            Convert.ToBase64String(authArray));
+        client.DefaultRequestHeaders.Add(
+            "user-agent",
+            "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.2;)");
+
+        var repoName = product.RepoBaseUrl.Split('/').Last();
+
+        // https://docs.github.com/en/rest/issues/issues#create-an-issue
+        var response = await client.PostAsyncAsJson($"repos/dotnet/{repoName}/issues", JsonConvert.SerializeObject(
+            new
+            {
+                title = $"[Automated] PRs inserted in VS build {vsBuildNumber}",
+                body = issueBody,
+                labels = new string[] { "vs-insertion" }
+            }));
+
+        if (!response.IsSuccessStatusCode)
+        {
+            logger.LogInformation($"Issue creation failed with status code: {response.StatusCode}");
```

Comment:
Maybe LogWarning or LogError.
```suggestion
            logger.LogError($"Issue creation failed with status code: {response.StatusCode}");
```